runtime-sdk/src/modules/evm/mod.rs

```diff
@@ -0,0 +1,238 @@
+        Self::do_evm(caller, ctx, |exec| {
+            let address = exec.create_address(evm::CreateScheme::Legacy { caller });
+            (
+                exec.transact_create(caller, U256::zero(), init_code, u64::max_value()),
+                address.as_bytes().to_vec(),
+            )
+        })
+    }
+
+    fn call<C: Context>(
+        ctx: &mut C,
+        caller: H160,
+        address: H160,
+        data: Vec<u8>,
+    ) -> Result<Vec<u8>, Error> {
+        Self::do_evm(caller, ctx, |exec| {
+            exec.transact_call(caller, address, U256::zero(), data, u64::max_value())
+        })
+    }
+
+    fn peek_storage<C: Context>(ctx: &mut C, address: H160, index: H256) -> Result<Vec<u8>, Error> {
+        let store =
+            storage::PrefixStore::new(ctx.runtime_state(), &crate::modules::evm::MODULE_NAME);
+        let storages =
+            storage::PrefixStore::new(store, &crate::modules::evm::evm_backend::STORAGES);
+        let s = storage::TypedStore::new(storage::PrefixStore::new(storages, &address));
+
+        let result: H256 = s.get(&index).unwrap_or_default();
+
+        Ok(result.as_bytes().to_vec())
+    }
+
+    fn peek_code<C: Context>(ctx: &mut C, address: H160) -> Result<Vec<u8>, Error> {
+        let store =
+            storage::PrefixStore::new(ctx.runtime_state(), &crate::modules::evm::MODULE_NAME);
+        let codes = storage::TypedStore::new(storage::PrefixStore::new(
+            store,
+            &crate::modules::evm::evm_backend::CODES,
+        ));
+
+        Ok(codes.get(&address).unwrap_or_default())
+    }
+}
+
+impl Module {
+    const EVM_CONFIG: Config = Config::istanbul();
+
+    fn do_evm<C, F, V>(source: H160, ctx: &mut C, f: F) -> Result<V, Error>
+    where
+        F: FnOnce(
+            &mut StackExecutor<'static, MemoryStackState<'_, 'static, evm_backend::Backend<'_, C>>>,
+        ) -> (evm::ExitReason, V),
+        C: Context,
+    {
+        let vicinity = evm_backend::Vicinity {
+            gas_price: U256::zero(),
+            origin: source,
+        };
+
+        let mut backend = evm_backend::Backend::<'_, C>::new(vicinity, ctx);
+        let metadata = StackSubstateMetadata::new(u64::max_value(), &Self::EVM_CONFIG);
+        let stackstate = MemoryStackState::new(metadata, &backend);
+        let mut executor = StackExecutor::new(stackstate, &Self::EVM_CONFIG);
+
+        let (exit_reason, exit_value) = f(&mut executor);
+
+        // TODO: Gas accounting?
+
+        let (vals, logs) = executor.into_state().deconstruct();
+        backend.apply(vals, logs, true);
+
+        if exit_reason.is_succeed() {
+            Ok(exit_value)
+        } else {
+            Err(Error::EVMError(format!("{:?}", exit_reason)))
+        }
+    }
+
+    fn tx_create<C: TxContext>(ctx: &mut C, body: types::CreateTx) -> Result<Vec<u8>, Error> {
+        Self::create(ctx, H160::from_slice(&body.caller), body.init_code)
```

Comment:
Hm, so caller should be derived from the first transaction signer (similarly to how we define the caller), maybe by only working for simple Secp256k1 keys for now (returning an error otherwise), using the same derivation scheme as Ethereum.